sdk/storage/azure-storage-quickquery/src/main/java/com/azure/storage/quickquery/models/BlobQuickQuerySerialization.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.quickquery.models;
+
+/**
+ * Defines the input and output serialization for a blob quick query request.
+ * either {@link BlobQuickQueryJsonSerialization} or {@link BlobQuickQueryDelimitedSerialization}
+ */
+public abstract class BlobQuickQuerySerialization {
+
+    protected char recordSeparator;
+
+    /**
+     * Gets the record separator.
+     * @return the record separator.
+     */
+    public char getRecordSeparator() {
+        return recordSeparator;
+    }
+
+    abstract BlobQuickQuerySerialization setRecordSeparator(char recordSeparator);
+
```

Comment:
By looking on derivatives of BlobQuickQuerySerialization I wonder why setRecordSeparator is abstract - both overrides are exactly the same.
Also by looking on apache avro implementation it seems that record separator is format independent property. What's the scenario where implementation of this would be different?

Reply:
Maybe record separator could be independent input to QQ APIs ? It doesn't seem to be coupled to format (all avro-parsable content types must have some kind of separator I think).